### ISSUE TITLE
Docker installation and CREATE_TOKENS_FOR_USERS

### ISSUE THREAD

**Issue by brendan-pike** [RocketChat/rocketchat_nextcloud#15]
I'm new to docker but installed rocketchat via docker as it was recommended.  It's been a learning curve, seems the default mongo image version is deprecated so learning to upgrade that was fun.  So its all running well but I need to include the "CREATE_TOKENS_FOR_USERS=true" but am having trouble achieving this.

The example is the readme is snap specific, please let me know how to do this in an established docker container which doesn't appear to run systemd.

Thank you

**Comment by pierrelocus** [comment]
Hi,

I'm not using Docker for it, but do you use a docker-compose.yml file ?
If so, you can add the env variables like this :

```
rocketchat:
    ...
    environment:
      YOUR_ENV_VARIABLE=true
```

Keep me updated if you need further help :)

Regards

**Comment by brendan-pike** [comment]
Okay, I worked it out and confirmed that the env. is set by using docker container inspect and could see,                 `"CREATE_TOKENS_FOR_USERS=true",`
But when a user clicks the rocketchat in nextcloud menu they are just shown a login screen? I was expecting their RC account to be automatically created, what am I missing?

**Comment by pierrelocus** [comment]
Yes normally it should automatically register (if needed) + connect the user.
Did you create the userId / Token pair in Nextcloud settings -> rocket chat ?
If so, can you provide me with logs from Nextcloud ? (found in /path/to/nextcloud/data/nextcloud.log)

Thanks in advance,

We're gonna soon put a explanation video on how to set up from scratch, it'll perhaps help too.

**Comment by brendan-pike** [comment]
In the logs I see, "Undefined array key \"authToken\" at /var/www/nextcloud/apps/rocketchat_nextcloud/lib/Rocket/User.php#241"
I tried regenerating the Auto generate Token and User ID in RC settings area but nothing changed. I tried removing/reinstall  the rc app but it didn't help either.

**Comment by pierrelocus** [comment]
Hi @brendan-pike ,
I'm not forgetting about you, I'm checking code and a new release will be there by the end of the week.
May I ask you, are you working on a Local environment or on servers with ssl installed etc ?

The module wouldn't work on local environments (or there are perhaps tweaks to do - I don't know them).
Regards,

**Comment by brendan-pike** [comment]
Thanks @pierrelocus 
It is a local environment installation, ie. both nextcloud and RC are on the same server, I run Nginx and yes with SSL.

**Comment by brendan-pike** [comment]
Just a largely unrelated heads up, but I've decided to restart with a fresh RC install.

I've had a very frustrating start made worse it seems by poorly maintained docs, the official recommended docker install https://docs.rocket.chat/quick-start/installing-and-updating/rapid-deployment-methods/docker-and-docker-compose uses a deprecated Mongo 4.0 version which then requires a complex migration which is very unwelcoming for a new user.  I managed to do this, but so many hours of frustration for a base level user like myself.  I see there is some efforts to update this https://github.com/RocketChat/Rocket.Chat/pull/23426 but 5 months later and the recommended docker-compose.yml hasn't even been updated which is left me wondering what I should do.

sigh.
